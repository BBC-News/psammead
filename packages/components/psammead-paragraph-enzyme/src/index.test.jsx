import React from 'react';
import Enzyme from 'enzyme';
// Enzyme adapter provides compatibility with React 16.x:
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import Paragraph from './index';

// configure enzyme to use the adapter you want it to use:
Enzyme.configure({ adapter: new Adapter() });

const p = <Paragraph>This is text in a paragraph.</Paragraph>;

// This suite uses Jest as its test runner, and is thus run from the root level `npm test`:
describe('Mounting, shallow and render tests', () => {
  describe('Paragraph tested with Enzyme mount', () => {
    // not sure what this does
    it('should mount a paragraph without throwing an error', () => {
      const wrapper = Enzyme.mount(p);
      // .c0
      expect(wrapper.length).toBe(1);
    });
  });

  describe('Paragraph tested with Enzyme shallow render', () => {
    it('should render correctly with no props', () => {
      const component = Enzyme.shallow(<Paragraph />);
      expect(toJson(component)).toMatchSnapshot();
    });

    it('should render correctly with text', () => {
      const component = Enzyme.shallow(p);
      expect(toJson(component)).toMatchSnapshot();
    });
  });

  describe('Paragraph tested with Enzyme rendering to static HTML', () => {
    it('should render to static HTML', () => {
      expect(Enzyme.render(p).text()).toEqual('This is text in a paragraph.');
    });
  });
});
