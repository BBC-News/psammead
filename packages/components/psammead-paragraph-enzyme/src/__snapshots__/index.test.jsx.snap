// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mounting, shallow and render tests Paragraph tested with Enzyme shallow render should render correctly with no props 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "Paragraph-sc-19yhspj-0",
        "isStatic": true,
        "lastClassName": "epjUaG",
        "rules": Array [
          "color:",
          "#404040",
          ";font-family:",
          "ReithSansNewsRegular, Helvetica, Arial, sans-serif",
          ";padding-bottom:",
          "1rem",
          ";margin:0;",
          "
  font-size: 0.9375em;
  line-height: 1.25rem;

  @media (min-width: 20em) {
    font-size: 1em;
    line-height: 1.375rem;
  }
",
          ";",
        ],
      },
      "displayName": "Paragraph",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "Paragraph-sc-19yhspj-0",
      "target": "p",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`Mounting, shallow and render tests Paragraph tested with Enzyme shallow render should render correctly with text 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "Paragraph-sc-19yhspj-0",
        "isStatic": true,
        "lastClassName": "epjUaG",
        "rules": Array [
          "color:",
          "#404040",
          ";font-family:",
          "ReithSansNewsRegular, Helvetica, Arial, sans-serif",
          ";padding-bottom:",
          "1rem",
          ";margin:0;",
          "
  font-size: 0.9375em;
  line-height: 1.25rem;

  @media (min-width: 20em) {
    font-size: 1em;
    line-height: 1.375rem;
  }
",
          ";",
        ],
      },
      "displayName": "Paragraph",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "Paragraph-sc-19yhspj-0",
      "target": "p",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
>
  This is text in a paragraph.
</StyledComponent>
`;
